### PR TITLE
capi: Access to ExecutionContext members

### PR DESCRIPTION
**Pull request by gumb0**
Depends on #626.

### CODE REVIEW

File: include/fizzy/fizzy.h [hunk 1]
```diff
@@ -596,18 +596,31 @@ bool fizzy_find_exported_memory(
 bool fizzy_find_exported_global(
     FizzyInstance* instance, const char* name, FizzyExternalGlobal* out_global) FIZZY_NOEXCEPT;
 
+FizzyExecutionContext* fizzy_create_execution_context(int depth) FIZZY_NOEXCEPT;
+
+FizzyExecutionContext* fizzy_create_metering_execution_context(
```
Comment (axic) [inline include/fizzy/fizzy.h:601]:
```suggestion
FizzyExecutionContext* fizzy_create_metered_execution_context(
```

File: include/fizzy/fizzy.h [hunk 2]
```diff
@@ -596,18 +596,31 @@ bool fizzy_find_exported_memory(
...
     FizzyInstance* instance, const char* name, FizzyExternalGlobal* out_global) FIZZY_NOEXCEPT;
 
+FizzyExecutionContext* fizzy_create_execution_context(int depth) FIZZY_NOEXCEPT;
+
+FizzyExecutionContext* fizzy_create_metered_execution_context(
+    int depth, int64_t ticks) FIZZY_NOEXCEPT;
+
+void fizzy_free_execution_context(FizzyExecutionContext* ctx) FIZZY_NOEXCEPT;
+
+int* fizzy_get_execution_context_depth(FizzyExecutionContext* ctx) FIZZY_NOEXCEPT;
```
Comment (gumb0) [inline include/fizzy/fizzy.h:606]: Does this returning `int*` look fine, or would `_get`/`_set` methods be better?

**Comment by axic** [comment]
@gumb0 do you want to rebase this on #626?

**Comment by gumb0** [comment]
Rebased.

File: include/fizzy/fizzy.h (hunk 2 above)
Comment (axic) [reply include/fizzy/fizzy.h:606]: Not sure, need to try in Rust how to interact with that, but we could merge this as a first step anyway.

**Review by axic — COMMENTED** [review]
I think this looks good generally, but let's get #626 merged first.

**Comment by axic** [comment]
@gumb0 can you rebase?

File: lib/fizzy/capi.cpp [hunk 3]
```diff
@@ -706,14 +706,35 @@ size_t fizzy_get_instance_memory_size(FizzyInstance* instance) noexcept
...
+
 FizzyExecutionResult fizzy_execute(FizzyInstance* c_instance, uint32_t func_idx,
     const FizzyValue* c_args, FizzyExecutionContext* c_ctx) noexcept
 {
     auto* instance = unwrap(c_instance);
     const auto* args = unwrap(c_args);
     const auto result =
         (c_ctx == nullptr ? fizzy::execute(*instance, func_idx, args) :
-                            fizzy::execute(*instance, func_idx, args, unwrap(c_ctx)));
+                            fizzy::execute(*instance, func_idx, args, *unwrap(c_ctx)));
```
Comment (axic) [inline lib/fizzy/capi.cpp:747]: Was this a typo in the previous commit?

**Review by axic — APPROVED** [review]
Looks good to me otherwise.

**Pull request merged**